components/TextFields/src/MDCTextInputControllerLegacyDefault.m

```diff
@@ -991,6 +1002,17 @@ + (void)setNormalColorDefault:(UIColor *)normalColorDefault {
                             : MDCTextInputControllerLegacyDefaultNormalUnderlineColorDefault();
 }
 
+- (NSString *)placeholderText {
+  return _textInput.placeholder;
+}
+
+- (void)setPlaceholderText:(NSString *)placeholderText {
+  _textInput.placeholder = placeholderText;
```

Comment:
Consider checking for equality with a quick exit to avoid updatePlaceholderAnimationConstraints:

Reply:
Done